File: 2017/2/spec/runner.spec.js
import { parse } from "../lib/parser";
import { part1, part2 } from "../lib/runner";

describe("Runner", () => {
  test("part 1 passes tests cases", () => {
    let input = `5 1 9 5
                 7 5 3
                 2 4 6 8`
    expect(part1(parse(input))).toEqual(18)
  });

  test("part 2 passes tests cases", () => {
    let input = `5 9 2 8
             9 4 7 3
             3 8 6 5`
    expect(part2(parse(input))).toEqual(9)
  });
});

File: 2020/8/spec/runner_spec.rb
# frozen_string_literal: true

require "runner"
require "parser"

describe Runner do
  it "calculates the correct answer for the test input for part1" do
    expect(Runner.part_1(Parser.parse(sample_input))).to eq(5)
  end

  it "calculates the correct answer for the test input part2" do
    expect(Runner.part_2(Parser.parse(sample_input))).to eq(8)
  end

  def sample_input
    <<~INPUT.strip
    nop +0
    acc +1
    jmp +4
    acc +3
    jmp -3
    acc -99
    acc +1
    jmp -4
    acc +6
    INPUT
  end
end
